src/actor/Verifier.ts

```diff
@@ -182,13 +185,34 @@ async function verifyPublicPresentation(
   return { verified, claims: verified ? attestedClaims : [] }
 }
 
+/**
+ * Checks whether each value is either an instance of an Accumulator or a [[PublicAttesterIdentity]].
+ *
+ * @param values Either an array or undefined which should be checked.
+ * @param isAcc A boolean to display whether we check for Accumulators.
+ * @returns True if the supplied values are correct.
+ */
+function verificationCheck(
+  values:
+    | Array<gabi.Accumulator | PublicAttesterIdentity | undefined>
+    | undefined,
+  isAcc: boolean
+): boolean {
```

Comment:
you've written a type guard, so why not make it explicit for typescript: 
```suggestion
): values is Array<gabi.Accumulator | PublicAttesterIdentity> {
```
However it would be most useful if you'd split this in two different type guards - one for `values is Array<PublicAttesterIdentity>` and one for `values is Array<gabi.Accumulator>` - while also dropping the `isAcc` argument.

Reply:
That is correct, thanks. Fixedin 5db18c9. 

I tried to add a general array typeguard with generics but this does not work since type information are removed at compile time such that `instanceof T` (whereas `T` is some generic) does not exist at runtime.

**Example**

```typescript
function arrayTypeguard<T>(values: T[] | unknown): values is T[] {
  if (Array.isArray(values) && values.length) {
    return values.every((a) => a instanceof T)
  }
  return false
}
```

**Error**
```
'T' only refers to a type, but is being used as a value here.ts(2693)
```